keras_hub/src/models/gemma3/gemma3_backbone.py

```diff
@@ -0,0 +1,382 @@
+import keras
+from keras import ops
+
+from keras_hub.src.api_export import keras_hub_export
+from keras_hub.src.layers.modeling.reversible_embedding import (
+    ReversibleEmbedding,
+)
+from keras_hub.src.models.backbone import Backbone
+from keras_hub.src.models.gemma.rms_normalization import RMSNormalization
+from keras_hub.src.models.gemma3.gemma3_decoder_block import Gemma3DecoderBlock
+from keras_hub.src.models.gemma3.gemma3_interleave_embeddings import (
+    Gemma3InterleaveEmbeddings,
+)
+from keras_hub.src.models.gemma3.gemma3_vision_output_encoder import (
+    Gemma3VisionOutputEncoder,
+)
+from keras_hub.src.models.gemma3.gemma3_vit import Gemma3Vit
+
+
+@keras_hub_export("keras_hub.models.Gemma3Backbone")
+class Gemma3Backbone(Backbone):
+    """Gemma3 core network with hyperparameters.
+
+    This backbone implements the mixed-modality Gemma3 architecture. It
+    contains a Visual Transformer network, as well as text token embedding
+    layer, followed by a backend-agnostic concatenation operation to
+    construct a sequence of representations of mixed type embeddings (visual
+    and textual). Then, the concatenated sequence is passed through a series
+    of Mixed Modality Decoder Blocks. The returned value from calling this model
+    represents probabilistic values for output tokens.
+
+    For a higher-level object for text-generation,
+    see `keras_hub.models.Gemma3CausalLM`.
+
+    The default constructor gives a fully customizable, randomly initialized
+    Gemma3 model with any number of vit layers, heads, embedding
+    dimensions, and equivalent configuration for Gemma3 Decoder layers. To
+    load preset architectures and weights, use the `from_preset` constructor.
+
+    Args:
+        vocabulary_size: int. The size of the token vocabulary.
+        image_size: int. The resolution of the image in both width and height.
+            The input images must be square.
+        image_max_length: int. The maximum number of images per sample (padded).
```

Comment:
`max_images_per_prompt` or `max_images_per_sample`? might be slightly clearer

Reply:
Actually, I think the right way to do this would be to not include this in the backbone config at all.

Sequence length needs to be constant to avoid recompilation, but we don't hardcode it into the backbone. We use `None` as the input shape, and use `ops.shape` to get the actual shape that was passed in. This mean you don't need to recreate the backbone to call it on a different sequence length, which is a big deal.

We should do the same for padded image length.